go/cmd/dolt/dolt.go

```diff
@@ -612,7 +612,10 @@ func buildLateBinder(ctx context.Context, cwdFS filesys.Filesys, mrEnv *env.Mult
 	useBranch, hasBranch := apr.GetValue(cli.BranchParam)
 
 	if hasUseDb && hasBranch{
-		dbName, _ := dsess.SplitRevisionDbName(useDb)
+		dbName, branchNameInDb := dsess.SplitRevisionDbName(useDb)
+		if len(branchNameInDb)!=0 {
+			cli.PrintErrf("The branch name in %s will be ignored.\n",useDb)
```

Comment:
We should fail in this case. Return the error so the user needs to provide reasonable arguments. Warning messages get ignored/missed.

Reply:
I have change it to return error, but still when i run bats test its failing